lib/collection/src/update_handler.rs

```diff
@@ -328,10 +335,25 @@ impl UpdateHandler {
                                 stopped,
                             ) {
                                 // Perform some actions when optimization if finished
-                                Ok(result) => {
+                                Ok((is_optimized, segment_id)) => {
+                                    if let Some(segment_id) = segment_id {
+                                        if let Some(segment) = segments.read().get(segment_id) {
+                                            // Any kind of optimization could have triggered indexing
+                                            let segment_ref = segment.get();
+                                            let segment_info = segment_ref.read();
+
+                                            if segment_info.segment_type() == SegmentType::Indexed {
```

Comment:
Optimized doesn't necessary creates Indexed segments

Reply:
We only want to count number of points indexed (not optimized). Do you still see a problem?

Reply:
What if indexed threshold is too high and it will never create any indexed segments? Optimization will still running

Reply:
Hmm, didn't think about that situation.

Then it sounds like we want to count optimized points, rather than indexed points. :face_exhaling:

Reply:
Do you propose to rename the param?

Reply:
> Do you propose to rename the param?

If we don't count indexed points, yes.

Reply:
Right.